go/interpreter/lfvm/instructions_test.go

```diff
@@ -1147,3 +1147,151 @@ func TestGenericCreate_MaxInitCodeSizeIsNotCheckedBeforeShanghai(t *testing.T) {
 		})
 	}
 }
+
+func TestInstructions_EIP2929_staticGasCostForBerlinAndAfterIsZero(t *testing.T) {
+	ops := []OpCode{BALANCE, EXTCODECOPY, EXTCODEHASH, EXTCODESIZE, CALL, CALLCODE, DELEGATECALL, STATICCALL}
+	for _, op := range ops {
+		if getBerlinGasPriceInternal(op) != 0 {
+			t.Errorf("expected zero gas cost for %v", op)
+		}
+	}
+}
+
+func TestInstructions_EIP2929_dynamicGasCostForBerlinAndAfterReportsOutOfGas(t *testing.T) {
```

Comment:
```suggestion
func TestInstructions_EIP2929_ReturnsErrorWhenNotEnoughGasForDinamicCost(t *testing.T) {
```
EIP2929 already says Berlin or later